Src/WitsmlExplorer.Frontend/components/ContextMenus/CopyComponentsToServer.tsx

```diff
@@ -1,71 +1,89 @@
 import { Server } from "../../models/server";
-import AuthorizationService from "../../services/authorizationService";
-import ComponentService from "../../services/componentService";
-import JobService, { JobType } from "../../services/jobService";
-import ObjectService from "../../services/objectService";
-import { displayMissingObjectModal } from "../Modals/MissingObjectModals";
-import { displayReplaceModal } from "../Modals/ReplaceModal";
-import { menuItemText, pluralize } from "./ContextMenuUtils";
+import { menuItemText } from "./ContextMenuUtils";
 import NestedMenuItem from "./NestedMenuItem";
+import { copyComponentsToServer } from "./CopyComponentsToServerUtils";
+import CopyRangeModal, { CopyRangeModalProps } from "../Modals/CopyRangeModal";
+import OperationType from "../../contexts/operationType";
 
 export interface CopyComponentsToServerMenuItemProps {
   componentsToCopy: { uid: string }[];
   componentType: ComponentType;
+  withRange?: boolean;
+}
+
+interface ComponentWithRange {
+  uid: string;
+  minIndex: number | Date;
+  maxIndex: number | Date;
 }
 
 export const CopyComponentsToServerMenuItem = (
   props: CopyComponentsToServerMenuItemProps
 ): React.ReactElement => {
-  const { componentsToCopy, componentType } = props;
+  const { componentsToCopy, componentType, withRange } = props;
   const {
     navigationState: { selectedServer, selectedObject, servers }
   } = useContext(NavigationContext);
   const { dispatchOperation } = useContext(OperationContext);
+  const menuComponents = menuItemText("copy", componentType, componentsToCopy);
+  const menuText =
+    withRange === true
+      ? menuComponents + " with range to server"
+      : menuComponents + " to server";
 
   return (
     <NestedMenuItem
       key={"copyComponentToServer"}
-      label={`${menuItemText(
-        "copy",
-        componentType,
-        componentsToCopy
-      )} to server`}
+      label={menuText}
       disabled={componentsToCopy.length < 1}
     >
       {servers.map(
         (server: Server) =>
           server.id !== selectedServer.id && (
             <MenuItem
               key={server.name}
-              onClick={() =>
-                copyComponentsToServer({
-                  targetServer: server,
-                  sourceServer: selectedServer,
-                  componentsToCopy,
-                  dispatchOperation,
-                  sourceParent: selectedObject,
-                  componentType
-                })
-              }
+              onClick={() => {
+                if (withRange === true) {
+                  const copyRangeModalProps: CopyRangeModalProps = {
+                    mnemonics: [],
+                    onSubmit(startIndex, endIndex) {
+                      const componentsToCopyWithRange =
+                        componentsToCopy as ComponentWithRange[];
+                      const componentsRange = componentsToCopyWithRange.filter(
```

Comment:
I think you might have misunderstood what they actually want to be copied. Right now, you are copying only mnemonics that are fully within the selected range. However, you want to copy only the data of each mnemonic that is within the selected range. So you only want to filter away mnemonics that lies completely outside the selected range. 

You also need to provide the start and end index in the call to the API.
This is from your code:
![image](https://github.com/equinor/witsml-explorer/assets/70512270/a96fa949-7099-43a5-baf4-c9b864187d7a)

This is how it looks like if you use the "paste"-functionality after copying a mnemonic with range:
![image](https://github.com/equinor/witsml-explorer/assets/70512270/6448ec9e-1eb4-4fc2-aad7-099083208b3b)

Reply:
solved